xmodule/assets/video/public/js/09_events_plugin.js

```diff
@@ -0,0 +1,177 @@
+        }
+    },
+
+    onPause: function() {
+        this.log('pause_video', {currentTime: this.getCurrentTime()});
+        this.emitPlayVideoEvent = true;
+    },
+
+    onComplete: function() {
+        this.log('complete_video', {currentTime: this.getCurrentTime()});
+    },
+
+    onEnded: function() {
+        this.log('stop_video', {currentTime: this.getCurrentTime()});
+        this.emitPlayVideoEvent = true;
+    },
+
+    onSkip: function(event, doNotShowAgain) {
+        let info = {currentTime: this.getCurrentTime()};
+        let eventName = doNotShowAgain ? 'do_not_show_again_video' : 'skip_video';
+        this.log(eventName, info);
+    },
+
+    onSeek: function(event, time, oldTime, type) {
+        this.log('seek_video', {
+            old_time: oldTime,
+            new_time: time,
+            type: type
+        });
+        this.emitPlayVideoEvent = true;
+    },
+
+    onSpeedChange: function(event, newSpeed, oldSpeed) {
+        this.log('speed_change_video', {
+            current_time: this.getCurrentTime(),
+            old_speed: this.state.speedToString(oldSpeed),
+            new_speed: this.state.speedToString(newSpeed)
+        });
+    },
+
+    onAutoAdvanceChange: function(event, enabled) {
+        this.log('auto_advance_change_video', {
+            enabled: enabled
+        });
+    },
+
+    onShowLanguageMenu: function() {
+        this.log('edx.video.language_menu.shown');
+    },
+
+    onHideLanguageMenu: function() {
+        this.log('edx.video.language_menu.hidden', {language: this.getCurrentLanguage()});
+    },
+
+    onShowTranscript: function() {
+        this.log('show_transcript', {current_time: this.getCurrentTime()});
+    },
+
+    onHideTranscript: function() {
+        this.log('hide_transcript', {current_time: this.getCurrentTime()});
+    },
+
+    onShowCaptions: function() {
+        this.log('edx.video.closed_captions.shown', {current_time: this.getCurrentTime()});
+    },
+
+    onHideCaptions: function() {
+        this.log('edx.video.closed_captions.hidden', {current_time: this.getCurrentTime()});
+    },
+
+    getCurrentTime: function() {
+        let player = this.state.videoPlayer;
+        let startTime = this.state.config.startTime;
+        let currentTime = player ? player.currentTime : 0;
+        // if video didn't start from 0(it's a subsection of video), subtract the additional time at start
+        if (startTime) {
+            currentTime = currentTime ? currentTime - startTime : 0;
+        }
+        return currentTime;
+    },
+
+    getCurrentLanguage: function() {
+        let language = this.state.lang;
+        return language;
+    },
+
+    log: function(eventName, data) {
+        // use startTime and endTime to calculate the duration to handle the case where only a subsection of video is used
+        let endTime = this.state.config.endTime || this.state.duration;
+        let startTime = this.state.config.startTime;
```

Comment:
We need to fix this like

`let startTime = this.state.config.startTime || 0; // Default to 0 if startTime is undefined
`

There is test in video_events_plugin_spec which test the scenario when startTime and endTime to be undefined, the code returning NaN in this case which should be fixed.

Test case:
https://github.com/salman2013/edx-platform/blob/8ecfb9bfa2723c015e3341491f37c84b8220405b/xmodule/js/spec/video/video_events_plugin_spec.js#L276